### PR TITLE
Add Schema Compare support for AAD

### PR DESCRIPTION
**Pull request by kisantia**
Adds support for AAD in schema compare by passing the AzureAcountToken if there is one. Fixes https://github.com/microsoft/azuredatastudio/issues/5123

### CODE REVIEW

**Comment by udeeshagautam** [comment]
(weird issue not letting me add comment in code)

SchemacompareUtils.cs line 121 : 
while using connInfo.ConnectionDetails.AzureAccountToken do we need to null check connInfo.ConnectionDetails also or this is being null checked before already?

**Comment by udeeshagautam** [comment]
(also not letting me edit the comment :( )

One more thing, I noticed one SC test failed in PR build... please confirm locally that all SC tests are passing with the change

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareUtils.cs [hunk 1]
```diff
@@ -117,7 +118,7 @@ internal static SchemaCompareEndpoint CreateSchemaCompareEndpoint(SchemaCompareE
                     }
                 case SchemaCompareEndpointType.Database:
                     {
-                        return new SchemaCompareDatabaseEndpoint(connectionString);
+                        return connInfo.ConnectionDetails.AzureAccountToken != null ? new SchemaCompareDatabaseEndpoint(connectionString, new AccessTokenProvider(connInfo.ConnectionDetails.AzureAccountToken)) : new SchemaCompareDatabaseEndpoint(connectionString);
                     }
```
Comment (pensivebrian) [inline src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareUtils.cs]: nit: Can you split this to multiple lines to be more readable?

**Comment by kisantia** [comment]
@udeeshagautam added null check and the test passed when I ran it locally with the changes.

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareUtils.cs [hunk 2]
```diff
@@ -118,7 +118,8 @@ internal static SchemaCompareEndpoint CreateSchemaCompareEndpoint(SchemaCompareE
...
                 case SchemaCompareEndpointType.Database:
                     {
-                        return connInfo.ConnectionDetails.AzureAccountToken != null ? new SchemaCompareDatabaseEndpoint(connectionString, new AccessTokenProvider(connInfo.ConnectionDetails.AzureAccountToken)) : new SchemaCompareDatabaseEndpoint(connectionString);
+                        return connInfo.ConnectionDetails != null && connInfo.ConnectionDetails.AzureAccountToken != null ?
```
Comment (pensivebrian) [inline src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareUtils.cs]: nit: I would write this like

```
return connInfo.ConnectionDetails?.AzureAccountToken != null 
    ? new SchemaCompareDatabaseEndpoint(connectionString, new AccessTokenProvider(connInfo.ConnectionDetails.AzureAccountToken)) 
    : new SchemaCompareDatabaseEndpoint(connectionString);
```

**Pull request merged**